### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by noxan** [ourownstory/neural_prophet#767]
# Documentation Structure

### Getting started (quickstart)
- Installation
- Basic model setup
- Basic forecasting and plotting

### Quick Tutorials (learn): quick rewards, minimal know how to use, give people overview
- Model components
- Plotting, Forecasting, evaluating

### How To Guides (specific): step by step, specific use case, how to solve a specific problem, advanced skills of using NeuralProphet
- The “real” how to guides: all on the same level
    - Migration (p-NP)
    - Data preparation
    - Hyperparameters
    - Validation and testing
- Application examples
    - …

### The science behind (understanding)
- Model architecture
- Presentation
- Link to paper, extracts

### API Reference
- User-facing (forecaster and utils) (emphasize)
- Complete (all)

### Community
- Slack
- Github
- Contributing.md

# Todos

Here goes a draft on how to restructure the documentation. Below there is a draft and a list of further todos.

- [ ] Should we add a welcome page like on github? I'd propose to move the development / contribution instructions to a separate page
- [ ] Should we make it possible to plot only selected components? (e.g. only trend, seasonality, etc). Could be interesting for the tutorials to reduce complexity
- [ ] Show strength and weaknesses of neuralprophet (e.g. in comparison to other models)
- [ ] Overall goals: Amount of explanatory text, code and visualizations should be balanced (approx. 1/3 each)

# Draft

## Getting started

Quickstart: Installation, minimal model, all auto, nice chart in the end, as few lines of code as possible

Incorporate into other sections: Dataset (electricity price might be interesting, very relatable, uses weather as input)

1. Basic model (and plots) - or should plots be a dedicated section? already contains components
2. Trend
3. Seasonality
4. Auto regression
5. Exogenous variables: Lagged and future regressors (atmospheric pressure, humidity, etc) - split in two sections?
6. Events and holidays (do not really connect to weather, might need other dataset?)
7. Uncertainty (TODO)
8. Global model (multiple datasets)
9. Validation (split dataset, forecast, performance metrics)
10. Recap, model overview (summary on components, link to details of neural model) and next steps

## Advanced topics / other topics

1. Model details (neural model, understanding the model, technical architecture)
2. Missing data: Interpolation
3. Normalization - is this a topic for the getting started section?
4. Hyperparameter tuning
5. Difference between multiplicative and additive components
6. Uncertainty: Confomal Prediction and Conformalized Quantile Regression

## New to time series?
Intro: Time series data: link to view resources? Kaggle inspiration

# Inspiration and examples
1. https://pytorch-lightning.readthedocs.io/en/latest/index.html
2. https://docs.djangoproject.com/en/4.1/
3. https://documentation.divio.com/

**Comment by noxan** [comment]
https://documentation.divio.com/ (thanks @ourownstory)

**Comment by noxan** [comment]
Feedback docs by Oskar

- Link existing notebooks (especially those on migrating from prophet)
- Consider renaming “the science behind section” to “for the scientist”

**Comment by noxan** [comment]
Feedback session, Dec 1st, 2022

Move tutorials to same folder as other jupyter notebooks.

## Tutorial 1

Put information about dataset right into the tutorial

Move dataset transforms to other repo

Some more information on model setup, like prophet, includes following components 

Extend forecast to future

Add description to what you see


## Tutorial 2

Remove epochs

Plot only trend (Leonie’s functions)

Start with default model

Explain default components of model in terms of trend

Explain results

Note: other components could have interaction, should ideally modeled together, see next sections
